src/Dapr.Workflow/WorkflowServiceCollectionExtensions.cs

```diff
@@ -153,31 +149,33 @@ static bool TryGetGrpcAddress(out string address)
             return false;
         }
 
-        static GrpcChannel CreateChannel(string address, HttpClient client)
+        static GrpcChannel CreateChannel(string address, HttpClient client, GrpcChannelOptions? grpcChannelOptions = null)
         {
-
-        GrpcChannelOptions options = new() { HttpClient = client};
-        var daprEndpoint = DaprDefaults.GetDefaultGrpcEndpoint();
-        if (!String.IsNullOrEmpty(daprEndpoint)) {
-            return GrpcChannel.ForAddress(daprEndpoint, options);
-        }
+            GrpcChannelOptions options = grpcChannelOptions ?? new();
+            options.HttpClient ??= client;
```

Comment:
A nit: if the caller provides gRPC options with a `HttpClient`, they won't get the Dapr API token behavior. If the caller does *not* provide a `HttpClient`, they'll only get access to "Dapr's" default `HttpClient` (after the fact, by virtue of it being attached to the options object).  Neither is great if the caller needs a more customized `HttpClient`.

While this is changed is patterned after the options handling in the `DaprClient` (for which I have the same nit), I wonder if it might be good to try to use a pattern that allows callers to change options but not at the risk of overriding customized behavior needed by Dapr.  A pattern that can then be replicated in `DaprClient`.  Is there any reason to not just expose a public `GrpcChannelOptions` property from `WorkflowRuntimeOptions`?  This property can be initialized to those settings typically used by Dapr (including the `HttpClient`).  The caller then has the ability to add/change settings as needed with less risk of stomping on Dapr-needed settings.

Reply:
please review again, changed the way the parameters are used.